### ISSUE TITLE
[inductor] `triton.runtime.jit` does not provide `get_cuda_stream`

### ISSUE THREAD

**Issue by ConnorBaker** [pytorch/pytorch#91827]
### 🐛 Describe the bug

Installing and using newer versions of Triton (either from master or from the `triton-mlir` branches) with Python 3.10 fails with an `ImportError` due to https://github.com/pytorch/pytorch/blob/94262efc7d381ace82aa74ed2f5f5ec76f8fca95/torch/_inductor/triton_ops/autotune.py#L27

`triton.runtime.jit` doesn't seem to make that symbol available. Instead, they use an `import as` to get a function of that same name from `torch` (this line is consistent across `master`, `legacy-backend`, and the new `triton-mlir` branch): https://github.com/openai/triton/blob/899bb0a0e7592612300913179d1a7d5baf5d4db5/python/triton/runtime/jit.py#L19

Inductor's codeine wrapper does as well: https://github.com/pytorch/pytorch/blob/94262efc7d381ace82aa74ed2f5f5ec76f8fca95/torch/_inductor/codegen/wrapper.py#L289

Should `autotune.py` be changed to match?

### Versions

```text
Collecting environment information...
PyTorch version: 2.0.0a0+git77c2a8a
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: 16.0.0 (++20230106042333+ea0cd51a4958-1~exp1~20230106042433.666)
CMake version: version 3.25.0
Libc version: glibc-2.35

Python version: 3.10.9+ (8af15cf:8af15cf, Jan  6 2023, 18:00:43) [Clang 16.0.0 (++20230106042333+ea0cd51a4958-1~exp1~20230106042433.666)] (64-bit runtime)
Python platform: Linux-6.0.15-300.fc37.x86_64-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4090
Nvidia driver version: 525.60.13
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.7.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.0
[pip3] torch==2.0.0a0+git77c2a8a
[pip3] torchvision==0.15.0a0+32d254b
[conda] Could not collect
```

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @mlazos @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

**Comment by ngimel** [comment]
Yes, we should change `autotune.py` as long as it keeps working with pre-MLIR triton, as we can't migrate to MLIR version yet.

**Comment by ConnorBaker** [comment]
> as we can't migrate to MLIR version yet.

Any reasons in particular?

I remember from Triton’s issue tracker that they’re fixing a few outstanding performance regressions since the backend rewrite. Speaking of which, do you all have a suite of regression tests for dynamo/torch.compile related stuff? Anything which could provide an additional datapoint for pre/post rewrite?

**Comment by ngimel** [comment]
The perf of post-MLIR version is not on par with pre-mlir version yet, we are working with @ptillet and @Jokeren to address it. We are using benchmarking suite in https://github.com/pytorch/torchdynamo/issues/681 to track performance, but so far even a simple Bert model is slowed down (there have been a few issues fixed, but some remain).

**Comment by ConnorBaker** [comment]
Is there are more easily digestible dashboard of that data somewhere?

And for that data, are you all using whatever was checking into the `master` branch of Triton when the build ran, or is it locked to the pinned commit in PyTorch?

**Comment by ngimel** [comment]
It's locked to the pinned commit in pytorch

**Comment by eellison** [comment]
Old version incompatibility issue, fixed by now i think.

**Comment by attashe** [comment]
@eellison 
Still have this issue with ControlNet training example

```bash
!accelerate launch train_controlnet.py \
 --pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5" \
 --output_dir="model_out" \
 --cache_dir="./cache" \
 --dataset_name=fusing/fill50k \
 --resolution=512 \
 --learning_rate=1e-5 \
 --validation_image "./conditioning_image_1.png" "./conditioning_image_2.png" \
 --validation_prompt "red circle with blue background" "cyan circle with brown floral background" \
 --train_batch_size=1 \
 --num_train_epochs=3 \
 --tracker_project_name="controlnet" \
 --enable_xformers_memory_efficient_attention \
 --checkpointing_steps=5000 \
 --validation_steps=5000 \
 --gradient_accumulation_steps=4 \
 --gradient_checkpointing \
 --set_grads_to_none
```

```python
home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/accelerator.py:384: UserWarning: `log_with=tensorboard` was passed but no supported trackers are currently installed.
  warnings.warn(f"`log_with={log_with}` was passed but no supported trackers are currently installed.")
12/18/2023 23:45:18 - INFO - __main__ - Distributed environment: DistributedType.NO
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda

Mixed precision type: bf16

You are using a model of type clip_text_model to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
{'clip_sample_range', 'prediction_type', 'thresholding', 'variance_type', 'dynamic_thresholding_ratio', 'timestep_spacing', 'sample_max_value'} was not found in config. Values will be initialized to default values.
{'scaling_factor', 'force_upcast'} was not found in config. Values will be initialized to default values.
{'cross_attention_norm', 'resnet_out_scale_factor', 'projection_class_embeddings_input_dim', 'transformer_layers_per_block', 'resnet_skip_time_act', 'mid_block_only_cross_attention', 'upcast_attention', 'num_attention_heads', 'time_embedding_act_fn', 'time_cond_proj_dim', 'reverse_transformer_layers_per_block', 'class_embeddings_concat', 'addition_time_embed_dim', 'time_embedding_type', 'conv_out_kernel', 'timestep_post_act', 'class_embed_type', 'use_linear_projection', 'encoder_hid_dim', 'addition_embed_type', 'dual_cross_attention', 'dropout', 'only_cross_attention', 'addition_embed_type_num_heads', 'time_embedding_dim', 'encoder_hid_dim_type', 'num_class_embeds', 'mid_block_type', 'conv_in_kernel', 'attention_type', 'resnet_time_scale_shift'} was not found in config. Values will be initialized to default values.
12/18/2023 23:45:19 - INFO - __main__ - Initializing controlnet weights from unet
Traceback (most recent call last):
  File "/home/attashe/Python Scripts/train_controlnet.py", line 1131, in <module>
    main(args)
  File "/home/attashe/Python Scripts/train_controlnet.py", line 911, in main
    controlnet, optimizer, train_dataloader, lr_scheduler = accelerator.prepare(
                                                            ^^^^^^^^^^^^^^^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/accelerator.py", line 1213, in prepare
    result = tuple(
             ^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/accelerator.py", line 1214, in <genexpr>
    self._prepare_one(obj, first_pass=True, device_placement=d) for obj, d in zip(args, device_placement)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/accelerator.py", line 1094, in _prepare_one
    return self.prepare_model(obj, device_placement=device_placement)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/accelerator.py", line 1409, in prepare_model
    model = torch.compile(model, **self.state.dynamo_plugin.to_kwargs())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/__init__.py", line 1723, in compile
    return torch._dynamo.optimize(backend=backend, nopython=fullgraph, dynamic=dynamic, disable=disable)(model)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_dynamo/eval_frame.py", line 610, in optimize
    compiler_config=backend.get_compiler_config()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/__init__.py", line 1571, in get_compiler_config
    from torch._inductor.compile_fx import get_patched_config_dict
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/compile_fx.py", line 38, in <module>
    from .fx_passes.joint_graph import joint_graph_passes
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/fx_passes/joint_graph.py", line 8, in <module>
    from ..pattern_matcher import (
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/pattern_matcher.py", line 28, in <module>
    from .lowering import fallback_node_due_to_unsupported_type
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/lowering.py", line 4768, in <module>
    import_submodule(kernel)
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_dynamo/utils.py", line 1492, in import_submodule
    importlib.import_module(f"{mod.__name__}.{filename[:-3]}")
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/kernel/bmm.py", line 4, in <module>
    from ..select_algorithm import (
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/select_algorithm.py", line 25, in <module>
    from .codegen.triton import texpr, TritonKernel, TritonPrinter, TritonScheduling
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/codegen/triton.py", line 26, in <module>
    from ..triton_heuristics import AutotuneHint
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/torch/_inductor/triton_heuristics.py", line 43, in <module>
    from triton.runtime.jit import get_cuda_stream, KernelInterface
ImportError: cannot import name 'get_cuda_stream' from 'triton.runtime.jit' (/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/triton/runtime/jit.py)
Traceback (most recent call last):
  File "/home/attashe/miniconda3/envs/pytorch/bin/accelerate", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/commands/accelerate_cli.py", line 47, in main
    args.func(args)
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/commands/launch.py", line 1017, in launch_command
    simple_launcher(args)
  File "/home/attashe/miniconda3/envs/pytorch/lib/python3.11/site-packages/accelerate/commands/launch.py", line 637, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['/home/attashe/miniconda3/envs/pytorch/bin/python', 'train_controlnet.py', '--pretrained_model_name_or_path=runwayml/stable-diffusion-v1-5', '--output_dir=model_out', '--cache_dir=./cache', '--dataset_name=fusing/fill50k', '--resolution=512', '--learning_rate=1e-5', '--validation_image', './conditioning_image_1.png', './conditioning_image_2.png', '--validation_prompt', 'red circle with blue background', 'cyan circle with brown floral background', '--train_batch_size=1', '--num_train_epochs=3', '--tracker_project_name=controlnet', '--enable_xformers_memory_efficient_attention', '--checkpointing_steps=5000', '--validation_steps=5000', '--gradient_accumulation_steps=4', '--gradient_checkpointing', '--set_grads_to_none']' returned non-zero exit status 1.
```

**Comment by netw0rkf10w** [comment]
@ngimel Sorry but why is this issue closed? The latest version (2.2.0) still has this issue. The following script (taken from here: https://pytorch.org/tutorials/recipes/compiling_optimizer.html) allows reproducing:

```python
import torch
import torch.utils.benchmark as benchmark

# exit cleanly if we are on a device that doesn't support torch.compile
if torch.cuda.get_device_capability() < (7, 0):
    print("Exiting because torch.compile is not supported on this device.")
    import sys
    sys.exit(0)


# Let's define a helpful benchmarking function:
def benchmark_torch_function_in_microseconds(f, *args, **kwargs):
    t0 = benchmark.Timer(
        stmt="f(*args, **kwargs)", globals={"args": args, "kwargs": kwargs, "f": f}
    )
    return t0.blocked_autorange().mean * 1e6

    
def main():
    model = torch.nn.Sequential(
        *[torch.nn.Linear(1024, 1024, False, device="cuda") for _ in range(10)]
    )
    input = torch.rand(1024, device="cuda")
    output = model(input)
    output.sum().backward()

    opt = torch.optim.Adam(model.parameters(), lr=0.01)

    @torch.compile(fullgraph=False)
    def fn():
        opt.step()

    # Warmup runs to compile the function
    for _ in range(5):
        fn()

    eager_runtime = benchmark_torch_function_in_microseconds(opt.step)
    compiled_runtime = benchmark_torch_function_in_microseconds(fn)

    assert eager_runtime > compiled_runtime

    print(f"eager runtime: {eager_runtime}us")
    print(f"compiled runtime: {compiled_runtime}us")

if __name__ == '__main__':
    main()

```

Traceback:

```bash
(python-3.10.4) bash-4.4$ python benchmark_optimizers.py 
Traceback (most recent call last):
  File "/home/user/code/visionBERT2/tests/benchmark_optimizers.py", line 46, in <module>
    main()
  File "/home/user/code/visionBERT2/tests/benchmark_optimizers.py", line 30, in main
    def fn():
  File "/home/user/.local/lib/python3.10/site-packages/torch/__init__.py", line 1802, in fn
    return compile(model,
  File "/home/user/.local/lib/python3.10/site-packages/torch/__init__.py", line 1820, in compile
    return torch._dynamo.optimize(backend=backend, nopython=fullgraph, dynamic=dynamic, disable=disable)(model)
  File "/home/user/.local/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 782, in optimize
    compiler_config=backend.get_compiler_config()
  File "/home/user/.local/lib/python3.10/site-packages/torch/__init__.py", line 1665, in get_compiler_config
    from torch._inductor.compile_fx import get_patched_config_dict
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 49, in <module>
    from .fx_passes.joint_graph import joint_graph_passes
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/fx_passes/joint_graph.py", line 12, in <module>
    from ..pattern_matcher import (
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/pattern_matcher.py", line 44, in <module>
    from .lowering import fallback_node_due_to_unsupported_type
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/lowering.py", line 5394, in <module>
    from . import kernel
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/kernel/__init__.py", line 1, in <module>
    from . import mm, mm_common, mm_plus_mm, unpack_mixed_mm
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/kernel/mm.py", line 7, in <module>
    from ..codegen.cuda.gemm_template import CUTLASSGemmTemplate
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/codegen/cuda/gemm_template.py", line 11, in <module>
    from .cuda_kernel import CUDATemplateKernel
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/codegen/cuda/cuda_kernel.py", line 7, in <module>
    from ...select_algorithm import ChoiceCaller
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/select_algorithm.py", line 24, in <module>
    from .codegen.triton import texpr, TritonKernel, TritonPrinter, TritonScheduling
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/codegen/triton.py", line 31, in <module>
    from ..triton_heuristics import AutotuneHint
  File "/home/user/.local/lib/python3.10/site-packages/torch/_inductor/triton_heuristics.py", line 54, in <module>
    from triton.runtime.jit import get_cuda_stream
ImportError: cannot import name 'get_cuda_stream' from 'triton.runtime.jit' (/home/user/.local/lib/python3.10/site-packages/triton/runtime/jit.py)
```
